scripts/library.sh

```diff
@@ -411,12 +411,29 @@ function start_knative_serving() {
   header "Starting Knative Serving"
   subheader "Installing Knative Serving"
   echo "Installing Serving CRDs from $1"
-  kubectl apply --selector knative.dev/crd-install=true -f "$1"
+  kubectl apply --selector knative.dev/crd-install=true -f "$1" || return 1
   echo "Installing the rest of serving components from $1"
-  kubectl apply -f "$1"
+  kubectl apply -f "$1" || return 1
   wait_until_pods_running knative-serving || return 1
 }
 
+# Install Knative Monitoring in the current cluster.
+# Parameters: $1 - Knative Monitoring manifest.
+function start_knative_monitoring() {
+  header "Starting Knative Monitoring"
+  subheader "Installing Knative Monitoring"
+  # namespace istio-system needs to be created first, due to the comment
+  # mentioned in
+  # https://github.com/knative/serving/blob/4202efc0dc12052edc0630515b101cbf8068a609/config/monitoring/tracing/zipkin/100-zipkin.yaml#L21
+  kubectl create namespace istio-system 2>/dev/null
+  echo "Installing Monitoring CRDs from $1"
+  kubectl apply --selector knative.dev/crd-install=true -f "$1" || return 1
```

Comment:
seems like this function is fast fail, can we use errexit here instead of return 1 everywhere?

Reply:
You don't want to do that, it's disruptive. We want the caller to decide what to do in case of failure.